### PR TITLE
Enhancement/delete pet

### PR DESCRIPTION
**Pull request by shjang1025**
- Add delete pet button and edit pet model for deleting all reminders when deleting specific pet

### CODE REVIEW

File: frontend/src/components/UpcomingReminder/UpcomingReminder.jsx [hunk 1]
```diff
@@ -48,10 +48,10 @@ const UpcomingReminder = () => {
 
     useEffect(() => {
         dispatch(fetchUserReminders(currentUser._id))
-    }, [dispatch, currentUser._id])
+    }, [dispatch, currentUser._id], upcomingReminders)
```
Comment (ChristopherJTrent) [inline frontend/src/components/UpcomingReminder/UpcomingReminder.jsx]: I've never seen this syntax before, what's the 3rd argument to useEffect do?

Comment (shjang1025) [reply frontend/src/components/UpcomingReminder/UpcomingReminder.jsx]: I think it was a typo. I will erase that and push it

Comment (ChristopherJTrent) [reply frontend/src/components/UpcomingReminder/UpcomingReminder.jsx]: Was it supposed to be like this?
```suggestion
    }, [dispatch, currentUser._id, upcomingReminders])
```

Comment (shjang1025) [reply frontend/src/components/UpcomingReminder/UpcomingReminder.jsx]: you can take upcomingReminders, so only dispatch, currentUser._id in the dependency array

Comment (ChristopherJTrent) [reply frontend/src/components/UpcomingReminder/UpcomingReminder.jsx]: Gotcha, will do.

**Review by ChristopherJTrent — APPROVED** [review]
LGTM

**Pull request merged**